web/massastation/src/layouts/LayoutStation/LayoutStation.tsx

```diff
@@ -1,83 +1,77 @@
 
 export interface LayoutStationProps {
   children?: ReactNode;
   navigator?: Navigator;
   onSetTheme?: () => void;
   storedTheme?: string;
-  activePage: string;
 }
 
-export function LayoutStation({ ...props }) {
-  const { children, navigator, onSetTheme, storedTheme, activePage } = props;
+interface NetworkRequest {
+  network: string;
+}
 
-  const navigate = useNavigate();
+export function LayoutStation({ ...props }) {
+  const { children, navigator, onSetTheme, storedTheme } = props;
 
   const [selectedTheme, setSelectedTheme] = useState(
     storedTheme || 'theme-dark',
   );
 
-  const searchIsActive = activePage === PAGES.SEARCH;
-
   function handleSetTheme(theme: string) {
     setSelectedTheme(theme);
 
     onSetTheme?.(theme);
   }
 
-  const { data: accounts = [] } = useResource<AccountObject[]>(
-    `${URL.WALLET_BASE_API}/${URL.WALLET_ACCOUNTS}`,
-  );
-
-  const currentAccount = useAccountStore((state) => state.currentAccount);
-  const setCurrentAccount = useAccountStore((state) => state.setCurrentAccount);
+  const { data: network, isSuccess: isSuccessNetwork } =
+    useResource<NetworkModel>(`${URL.PATH_NETWORKS}`);
+
+  const [
+    currentNetwork,
+    availableNetworks,
+    setCurrentNetwork,
+    setAvailableNetworks,
+  ] = useNetworkStore((state) => [
+    state.currentNetwork,
+    state.availableNetworks,
+    state.setCurrentNetwork,
+    state.setAvailableNetworks,
+  ]);
 
-  const accountsItems = accounts.map((account) => ({
-    icon: <Identicon username={account.nickname} size={32} />,
-    item: account.nickname,
-    onClick: () => setCurrentAccount(account.nickname),
-  }));
+  useEffect(() => {
+    if (isSuccessNetwork) {
+      if (network.currentNetwork) setCurrentNetwork(network.currentNetwork);
+      if (network.availableNetworks)
+        setAvailableNetworks(network.availableNetworks);
+    }
+  }, [isSuccessNetwork]);
 
-  const selectedAccountKey: number = parseInt(
-    Object.keys(accounts).find(
-      (_, idx) => accounts[idx].nickname === currentAccount,
+  const selectedNetworkKey: number = parseInt(
+    Object.keys(availableNetworks).find(
+      (_, idx) => availableNetworks[idx] === currentNetwork,
     ) || '0',
   );
 
-  const existingAccount: boolean = accounts.length > 0;
-
-  const [pluginWalletIsInstalled, setPluginWalletIsInstalled] = useState(false);
-
-  const { data: plugins, isSuccess } =
-    useResource<IMassaStore[]>('plugin-manager');
+  const { mutate: mutateUpdateNetwork } = usePost<NetworkRequest>(
+    `${URL.PATH_NETWORKS}/${currentNetwork}`,
+  );
 
-  useEffect(() => {
-    if (isSuccess) {
-      setPluginWalletIsInstalled(
-        plugins.some((plugin) => plugin.name === MASSA_WALLET),
-      );
-    }
-  }, [isSuccess, plugins]);
+  const availableNetworksItems = availableNetworks.map((network) => ({
+    item: network,
+    onClick: () => {
+      setCurrentNetwork(network);
+      mutateUpdateNetwork({});
+      window.location.reload();
```

Comment:
Is it really necessary? It will decrease UX quality?

Reply:
@Thykof  You talk about var `window.location.reload()` ?

Reply:
> you can use the refetch function from the useRessource if you want to update the infos

I call `mutateUpdateNetwork({});` to use Post method from API.